bitcoin/src/blockdata/constants.rs

```diff
@@ -103,9 +125,10 @@ fn bitcoin_genesis_tx() -> Transaction {
 
 /// Constructs and returns the genesis block.
 pub fn genesis_block(params: impl AsRef<Params>) -> Block {
-    let txdata = vec![bitcoin_genesis_tx()];
+    let txdata = vec![bitcoin_genesis_tx(params.as_ref())];
     let hash: sha256d::Hash = txdata[0].compute_txid().into();
-    let merkle_root = hash.into();
+    let merkle_root: crate::TxMerkleNode = hash.into();
+
     match params.as_ref().network {
```

Comment:
`as_ref()` can be a user implemented function, so only calling it once and storing the result in a local variable prevents people from getting `txdata` of one network and the rest of another network.

Reply:
Dear friend, thank you I have revised it according to your suggestion